src/simulation/dynamics/facetSRPDynamicEffector/facetSRPDynamicEffector.h

```diff
@@ -0,0 +1,68 @@
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+
+#ifndef FACET_SRP_DYNAMIC_EFFECTOR_H
+#define FACET_SRP_DYNAMIC_EFFECTOR_H
+
+#include <Eigen/Dense>
+#include <vector>
+#include "simulation/dynamics/_GeneralModuleFiles/dynamicEffector.h"
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/msgPayloadDefC/SpicePlanetStateMsgPayload.h"
+#include "simulation/dynamics/_GeneralModuleFiles/stateData.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/avsEigenMRP.h"
+
+/*! @brief Spacecraft Geometry Data */
+typedef struct {
+  std::vector<double> facetAreas;                                   //!< [m^2] Vector of facet areas
+  std::vector<double> facetSpecCoeffs;                              //!< Vector of facet spectral reflection optical coefficients
+  std::vector<double> facetDiffCoeffs;                              //!< Vector of facet diffuse reflection optical coefficients
+  std::vector<Eigen::Vector3d> facetNormals_B;                      //!< Vector of facet normals expressed in B frame components
+  std::vector<Eigen::Vector3d> facetLocationsPntB_B;                //!< [m] Vector of facet locations wrt point B in B frame components
+}FacetedSRPSpacecraftGeometryData;
+
+/*! @brief Faceted Solar Radiation Pressure Dynamic Effector */
+class FacetSRPDynamicEffector: public SysModel, public DynamicEffector
+{
+public:                                                             
+    FacetSRPDynamicEffector();                                      //!< The module constructor
+    ~FacetSRPDynamicEffector();                                     //!< The module destructor
+    void linkInStates(DynParamManager& states);                     //!< Method for giving the effector access to the hub states
+    void computeForceTorque(double integTime, double timeStep);     //!< Method for computing the SRP force and torque about point B
+    void Reset(uint64_t CurrentSimNanos);                           //!< Reset method
+    void UpdateState(uint64_t CurrentSimNanos);                     //!< Method for updating the effector states
+    void WriteOutputMessages(uint64_t CurrentClock);                //!< Method for writing the output messages
```

Comment:
:small_red_triangle_down: function names start lower case in Basilisk. Same with the parameter in the method prototype/definition. The ones that aren't are legacy, that one day will be changed :)

Reply:
Done.